app/code/Magento/MediaContentSynchronization/Test/Integration/Model/PublisherTest.php

```diff
@@ -0,0 +1,125 @@
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronization\Test\Integration\Model;
+
+use Magento\Framework\Exception\IntegrationException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\MessageQueue\ConsumerFactory;
+use Magento\MediaContentApi\Api\Data\ContentIdentityInterfaceFactory;
+use Magento\MediaContentApi\Api\GetAssetIdsByContentIdentityInterface;
+use Magento\MediaContentApi\Api\GetContentByAssetIdsInterface;
+use Magento\MediaContentSynchronization\Model\Publish;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test for media content Publisher
+ */
+class PublisherTest extends TestCase
+{
+    private const TOPIC_MEDIA_CONTENT_SYNCHRONIZATION = 'media.content.synchronization';
+
+    /**
+     * @var ConsumerFactory
+     */
+    private $consumerFactory;
+
+    /**
+     * @var ContentIdentityInterfaceFactory
+     */
+    private $contentIdentityFactory;
+
+    /**
+     * @var GetContentByAssetIdsInterface
+     */
+    private $getContentIdentities;
+
+    /**
+     * @var GetAssetIdsByContentIdentityInterface
+     */
+    private $getAssetIds;
+
+    /**
+     * @var Publish
+     */
+    private $publish;
+
+    protected function setUp(): void
+    {
+        $this->consumerFactory = Bootstrap::getObjectManager()->get(ConsumerFactory::class);
+        $this->contentIdentityFactory = Bootstrap::getObjectManager()->get(ContentIdentityInterfaceFactory::class);
+        $this->getContentIdentities = Bootstrap::getObjectManager()->get(GetContentByAssetIdsInterface::class);
+        $this->getAssetIds = Bootstrap::getObjectManager()->get(GetAssetIdsByContentIdentityInterface::class);
+        $this->publish = Bootstrap::getObjectManager()->get(Publish::class);
+    }
+
+    /**
+     * @dataProvider filesProvider
+     * @magentoDataFixture Magento/MediaContentCatalog/_files/category_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCatalog/_files/product_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCms/_files/page_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCms/_files/block_with_asset.php
+     * @magentoDataFixture Magento/MediaGallery/_files/media_asset.php
+     * @param array $contentIdentities
+     * @throws IntegrationException
+     * @throws LocalizedException
+     */
+    public function testExecute(array $contentIdentities): void
+    {
+        // publish message to the queue
+        $this->publish->execute($contentIdentities);
+
+        // run and process message
+        $batchSize = 1;
+        $maxNumberOfMessages = 1;
+        $consumer = $this->consumerFactory->get(self::TOPIC_MEDIA_CONTENT_SYNCHRONIZATION, $batchSize);
+        $consumer->process($maxNumberOfMessages);
+
+        // verify synchronized media content
+        foreach ($contentIdentities as $contentIdentity) {
+            $assetId = 2020;
```

Comment:
please see the comments in other tests and move out of the loop everything that can be executed outside of the loop